src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,49 +1,96 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+
+type Props = {
+  todo: Todo;
+  onClose: () => void;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  todo,
+  onClose,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [hasLoadingError, setHasLoadingError] = useState(false);
+  const {
+    id,
+    title,
+    completed,
+    userId,
+  } = todo;
+
+  const waitForUsers = async () => {
+    try {
+      const userFromServer = await getUser(userId);
+
+      setUser(userFromServer);
+    } catch {
+      setHasLoadingError(true);
+    } finally {
+      setHasLoadingError(true);
+    }
```

Comment:
Finally will always work, if you will catch error and then it will be finally case anyway. So your code won't work as expected